services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -117,59 +130,105 @@ async def _get_service_access_rights(
+    # Rationale: if a project template was published, its services must be available to everyone.
+    # a published template has a column Published that is set to True
+    projects_repo = ProjectsRepository(connection)
+    published_services: Set[Tuple[str, str]] = {
+        (service.key, service.version)
+        for service in await projects_repo.list_services_from_published_templates()
+    }
+
+    groups_repo = GroupsRepository(connection)
+    everyone_gid = (await groups_repo.get_everyone_group()).gid
+
+    services_repo = ServicesRepository(connection)
+    available_services: Set[Tuple[str, str]] = {
+        (service.key, service.version)
+        for service in await services_repo.list_services(
+            gids=[everyone_gid], execute_access=True
+        )
+    }
+
+    missing_services = published_services - available_services
+    missing_services_access_rights = [
+        ServiceAccessRightsAtDB(
+            key=service[0], version=service[1], gid=everyone_gid, execute_rights=True
+        )
+        for service in missing_services
+    ]
+    if missing_services_access_rights:
+        logger.info(
+            "Adding access rights for published templates\n: %s",
+            missing_services_access_rights,
+        )
+        await services_repo.upsert_service_access_rights(missing_services_access_rights)
 
 
 async def sync_registry_task(app: FastAPI) -> None:
     # get list of services from director
-    while True:
-        try:
-            logger.debug("syncing services between registry and database...")
-            services_in_registry: Dict[
-                Tuple[ServiceKey, ServiceVersion], ServiceDockerData
-            ] = await _list_registry_services(app)
-            services_in_db: Set[
-                Tuple[ServiceKey, ServiceVersion]
-            ] = await _list_db_services(app)
-
-            # check that the db has all the services at least once
-            missing_services_in_db = set(services_in_registry.keys()) - services_in_db
-            if missing_services_in_db:
-                logger.debug(
-                    "missing services in db:\n%s", pformat(missing_services_in_db),
-                )
-                # update db (rationale: missing services are shared with everyone for now)
-                await _create_services_in_db(
-                    app, missing_services_in_db, services_in_registry
-                )
-
-            await asyncio.sleep(app.state.settings.background_task_rest_time)
-
-        except CancelledError:
-            # task is stopped
-            return
-        except Exception:  # pylint: disable=broad-except
-            logger.exception("Error while processing services entry")
-            await asyncio.sleep(
-                5
-            )  # wait a bit before retrying, so it does not block everything until the director is up
+    engine: Engine = app.state.engine
+    async with engine.acquire() as conn:
```

Comment:
This is one of the discussions we had. Having the ``while True`` within the context manager, holds the db connection forever.
why not releasing it before going to sleep?? This way we give a chance to other requests to the database


```python
  while True:
     try:
         async with engine.acquire() as conn:
               _ensure_...(app, conn)
               _ensure_...(app, conn)
         await asyncio.sleep(...)
     except ...
```

Reply:
who are others? just kidding, it's a good point. will do.

Reply:
ok I thought we should acquire once but that makes more sense like this.